src/tomography.jl

```diff
@@ -0,0 +1,244 @@
+    else
+                Uset1Q[1]=eye(2);
+                Uset1Q[2]=expm(1im*acos(-1/3)*X);
+                Uset1Q[3]=expm(1im*acos(-1/3)*X)*expm(1im*2*pi/3*Z);
+                Uset1Q[4]=expm(1im*acos(-1/3)*X)*expm(-1im*2*pi/3*Z);
+    end
+  else
+    error("Invalid prep./meas. pulse pulse_type")
+  end
+elseif nbrPulses==6
+  #Six pulse set
+        Uset1Q[1]=eye(2);
+        Uset1Q[2]=expm(-1im*(pi/4)*X);
+        Uset1Q[3]=expm(1im*(pi/4)*X);
+        Uset1Q[4]=expm(-1im*(pi/4)*Y);
+        Uset1Q[5]=expm(1im*(pi/4)*Y);
+        Uset1Q[6]=expm(-1im*(pi/2)*X);
+elseif nbrPulses==12
+  #12 pulse set
+        Uset1Q[1] = I;
+        Uset1Q[2] = expm(-1im*(pi/2)*X);
+        Uset1Q[3] = expm(-1im*(pi/2)*Y);
+        Uset1Q[4] = expm(-1im*(pi/2)*Z);
+        Uset1Q[5] = expm(-1im*(pi/3)*(1/sqrt(3))*(X+Y-Z));  #X+Y-Z 120
+        Uset1Q[6] = expm(-1im*(pi/3)*(1/sqrt(3))*(X-Y+Z));  #X-Y+Z 120
+        Uset1Q[7] = expm(-1im*(pi/3)*(1/sqrt(3))*(-X+Y+Z));  #-X+Y+Z 120
+        Uset1Q[8] = expm(-1im*(pi/3)*(1/sqrt(3))*(-X-Y-Z));  #X+Y+Z -120 (equivalent to -X-Y-Z 120)
+        Uset1Q[9] = expm(-1im*(pi/3)*(1/sqrt(3))*(X+Y+Z));  #X+Y+Z 120
+        Uset1Q[10] = expm(-1im*(pi/3)*(1/sqrt(3))*(-X+Y-Z));  #X-Y+Z -120 (equivalent to -X+Y-Z 120)
+        Uset1Q[11] = expm(-1im*(pi/3)*(1/sqrt(3))*(X-Y-Z));  #-X+Y+Z -120 (equivalent to X-Y-Z 120
+        Uset1Q[12] = expm(-1im*(pi/3)*(1/sqrt(3))*(-X-Y+Z));  #X+Y-Z -120 (equivalent to -X-Y+Z 120
+else
+  error("Invalid number of pulses");
+end
+
+#Now kron things together
+#First create a matrix with giving the mod nbrPulses description of which 1Q gates to kron together
+numGates = nbrPulses^nbrQubits;
+kronMat = zeros(UInt8, numGates, nbrQubits);
+for qubitct = 1:nbrQubits
+        kronMat[:,qubitct] = reshape(repmat(transpose(1:nbrPulses), nbrPulses^(nbrQubits-qubitct), nbrPulses^(qubitct-1)), numGates, 1);
+end
+
+for gatect = 1:numGates
+        gateSet[gatect] = 1;
+  for qubitct = 1:nbrQubits
+            gateSet[gatect] = kron(gateSet[gatect], Uset1Q[kronMat[gatect, qubitct]]);
+  end
+end
+    return gateSet
+end
+
+function QST_LSQ(expResults, varMat, measPulseMap, measOpMap, measPulseUs, measOps, n)
+
+#Function to perform least-squares inversion of state tomography data
+#
+# expResults : data array
+# varmat : convariance matrix for data
+# measPulseMap: array mapping each experiment to a measurement readout
+# pulse
+# measOpMap: array mapping each experiment to a measurement channel
+# measPulseUs : cell array of unitaries of measurement pulses
+# measOps : cell array of measurement operators for each channel
+# n : number of qubits
+
+#Construct the predictor matrix.  Each row is an experiment.  The number of
+#columns is 4^n for the size of the vectorized density matrix.
+#%First transform the measurement operators by the readout pulses to create
+#the effective measurement operators and then flatten into row of the
+#predictor matrix
+@printf("Setting up predictor matrix....");
```

Comment:
For printing just a plain string, use `print` or `println` rather than `@printf`.